internal/controller/pipelinerollout_controller.go

```diff
@@ -177,8 +185,21 @@ func (r *PipelineRolloutReconciler) needsUpdate(old, new *apiv1.PipelineRollout)
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *PipelineRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		// Reconcile PipelineRollouts when there's been a Generation changed (i.e. Spec change)
-		For(&apiv1.PipelineRollout{}).WithEventFilter(predicate.GenerationChangedPredicate{}).
-		Complete(r)
+
+	controller, err := runtimecontroller.New(ControllerPipelineRollout, mgr, runtimecontroller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch PipelineRollouts
```

Comment:
Why do we need to watch PipelineRollouts specifically? I thought they are watched by default?

Reply:
they were being watched before in the code which this is replacing. It's just a different call

Reply:
For reference, I modeled this code on Derek's [PR](https://github.com/numaproj-labs/numaplane/pull/158/files/1148afd9fea5f766c99caa92e629527a5c15f470) which he had written by accident because he didn't realize I already had the code that was doing what he intended to do.